Source/WebInspectorUI/UserInterface/Views/SpreadsheetCSSStyleDeclarationSection.js

```diff
@@ -107,22 +107,29 @@ WI.SpreadsheetCSSStyleDeclarationSection = class SpreadsheetCSSStyleDeclarationS
             let groupingsElement = this.element.appendChild(document.createElement("div"));
             groupingsElement.classList.add("header-groupings");
 
-            let currentGroupingType = null;
-            let currentGroupingHadText = false;
-            let groupingTypeElement = null;
             this._groupingElements = groupings.map((grouping) => {
-                if (grouping.type !== currentGroupingType || !grouping.text || !currentGroupingHadText) {
-                    groupingTypeElement = groupingsElement.appendChild(document.createElement("div"));
-                    groupingTypeElement.classList.add("grouping");
-                    groupingTypeElement.textContent = grouping.prefix + " ";
-                    currentGroupingType = grouping.type;
-                } else
-                    groupingTypeElement.append(grouping.isLayer && grouping.text ? "." : ", ");
-
-                currentGroupingHadText = !!grouping.text;
-                let span = groupingTypeElement.appendChild(document.createElement("span"));
-                span.textContent = grouping.text ?? "";
-                return span;
+                let groupingTypeElement = groupingsElement.appendChild(document.createElement("div"));
+                groupingTypeElement.classList.add("grouping");
+                groupingTypeElement.textContent = grouping.prefix + " ";
+
+                let groupingTextElement = groupingTypeElement.appendChild(document.createElement("span"));
+                groupingTextElement.textContent = grouping.text ?? "";
+                groupingTextElement.representedGrouping = grouping;
+
+                if (grouping.editable) {
+                    let groupingTextField = new WI.SpreadsheetRuleHeaderField(this, groupingTextElement);
+                    groupingTextField.addEventListener(WI.SpreadsheetRuleHeaderField.Event.StartedEditing, function(event) {
```

Comment:
these should really be member functions so that we don't have to create a new instance of the same `function` each time we render a selector